src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorExtensions.cs

```diff
@@ -2779,7 +2821,33 @@ public static TensorSpan<T> Reshape<T>(in this TensorSpan<T> tensor, params scop
             nint tempLinear = TensorSpanHelpers.CalculateTotalLength(arrLengths);
             if (tempLinear != tensor.FlattenedLength)
                 ThrowHelper.ThrowArgument_InvalidReshapeDimensions();
-            nint[] strides = TensorSpanHelpers.CalculateStrides(arrLengths);
+
+            nint[] strides;
+
+            // If we contain a 0 stride we can only add dimensions of length 1.
+            if (tensor.Strides.Contains(0))
+            {
+                List<nint> origStrides = new List<nint>(tensor.Strides.ToArray());
+                int lengthOffset = 0;
+                for (int i = 0; i < arrLengths.Length; i++)
+                {
+                    if (lengthOffset < tensor.Rank && arrLengths[i] == tensor.Lengths[lengthOffset])
+                        lengthOffset++;
```

Comment:
nit: This violates the coding-style guidelines: https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md

> 18. When using a single-statement if, we follow these conventions:
* Never use single-line form (for example: if (source == null) throw new ArgumentNullException("source");)
* Using braces is always accepted, and required if any block of an if/else if/.../else compound statement uses braces or if a single statement body spans multiple lines.
* Braces may be omitted only if the body of every block associated with an if/else if/.../else compound statement is placed on a single line.

And my own person preference is to never skip braces, as skipping them can lead to trivial bugs, especially across refactorings.

Reply:
Good to know. I'll get the bradces added in.

Reply:
Fixed.